mapper/orm/src/main/java/org/hibernate/search/mapper/orm/mapping/impl/HibernateOrmTypeContextContainer.java

```diff
@@ -109,6 +123,17 @@ private HibernateOrmTypeContextContainer(Builder builder, SessionFactoryImplemen
 		this.byJpaEntityName = new KeyValueProvider<>( byJpaEntityNameContent, log::unknownJpaEntityNameForMappedEntityType );
 		this.indexedByJpaEntityName = new KeyValueProvider<>( indexedByJpaEntityNameContent, log::unknownJpaEntityNameForIndexedEntityType );
 		this.byHibernateOrmEntityName = new KeyValueProvider<>( byHibernateOrmEntityNameContent, log::unknownHibernateOrmEntityNameForMappedEntityType );
+		this.indexedWithSuperTypesByExactClass = new KeyValueProvider<>( indexedWithSuperTypesByExactClassContent, log::unknownClassForMappedEntityType );
+	}
+
+	private boolean isManagedType(SessionFactoryImplementor sessionFactory, Class<?> javaClass) {
+		try {
+			sessionFactory.getMetamodel().managedType( javaClass );
+		}
+		catch (Exception e) {
+			return false;
+		}
+		return true;
```

Comment:
FYI in case you need it: this could be implemented in the `pojo-base` module, just using different metadata (`org.hibernate.search.mapper.pojo.model.additionalmetadata.impl.PojoTypeAdditionalMetadata#isEntity`, which does cover all Hibernate ORM entities as well).

Reply:
Thanks. I think I've tried this method at some point, but it would return false for a mapped supertype. I'll see how that change for the types goes 😃